components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -3076,28 +3093,54 @@ def buildQuery(self, obj_type, ids=None, params=None, attributes=None):
                 "getObjects uses a string to define obj_type, E.g. "
                 "'Image' not %r" % obj_type)
 
-        if params is None:
-            params = omero.sys.Parameters()
-        if params.map is None:
-            params.map = {}
+        owner = None
+        order_by = None
+        opts = None
+        inputParams = None
+
+        if isinstance(params, dict):
```

Comment:
I'm really not sure about this. At a minimum it needs documentation and to detailed review in the changelog as above. While it certainly keeps the prototype is it not feasible to add `opts` as an additional keyword argument?